doc/specs/#500 - Auto generate manifests for newer versions.md

```diff
@@ -0,0 +1,87 @@
+---
+author: Vedant Mohan Goyal @vedantmgoyal9
+created on: 2025-01-25
+last updated: 2025-03-09
+issue id: https://github.com/microsoft/winget-pkgs/issues/500
+---
+
+# Auto-generate Manifests for Newer Versions
+
+## Abstract
+
+This spec outlines the design for automation that generates manifests for new versions of applications in the Windows Package Manager Community Repository (winget-pkgs). It describes a meta-manifest that provides the necessary information (e.g., version, installer URLs, release date, and notes) to automate the creation of new manifests.
+
+## Problem Statement
+
+Currently, winget-pkgs relies on community contributors to update package versions, leading to delays between a new software release and its availability in WinGet. This delay is problematic when updates address security vulnerabilities, leaving users exposed. Faster updates would make WinGet more reliable, particularly for businesses that depend on timely and verified application availability.
+
+## Solution Design
+
+The proposed automation will automatically open pull requests (PRs) for new versions of applications when a new release is detected via the software’s API.
+
+```jsonc
+{
+  "$schema": "<link to schema>",
+  "PackageIdentifier": "AdGuard.AdGuard",
+  "Source": [
+    {
+      "InvokeType": "RestMethod",
+      "Uri": "",
+      "Method": "Get",
+      // other switches and args supported by Invoke-RestMethod/WebRequestHeaders
+      // like Body, UserAgent, MaximumRedirection, SkipHttpErrorCheck, ErrorAction
+    }
+  ],
+  "PostResponseScript": "$Response = $Response.response",
+  "Update": {
+    "PackageVersion": "$Response.version.'#cdata-section'",
+    "InstallerUrls": "$Response.'update-url'.'#cdata-section'"
+  },
+  "AdditionalInfo": {
+    "ForceUpgrade": false
+  },
+  "Skip": {
+    "Skip?": false,
+    "Reason": null
+  }
+}
+```
+
+The above JSON defines automation metadata and will be stored in a `.auto_updates.json` file alongside `.validation` files.
+
+### Automation Workflow
+
+The automation consists of a PowerShell script that:
+
+1. Loads variables specified in the `AdditionalInfo` object.
+2. Executes `Invoke-RestMethod` or `Invoke-WebRequest` with the defined arguments (supporting multiple sources for different APIs).
+3. Runs the `PostResponseScript` to process API responses.
+4. Calls `wingetcreate update` in non-interactive mode:
+   - Uses `--submit` if no additional metadata is required.
+   - If additional metadata is needed, modifies the manifests manually before calling `wingetcreate submit`.
+
+This PowerShell script will be executed in an Azure Pipeline on a set schedule.
+
+## UI/UX Design
```

Comment:
What is the UI / UX of the PRs that are submitted using the automation? Do they appear as normal PRs? Is there something linking them to the automation run? Are there any logs that are generated?